internal/searchindex/searchindex.go

```diff
@@ -0,0 +1,440 @@
+package searchindex
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/compat"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+)
+
+type SearchIndex struct {
+	akov2.SearchIndex
+	akov2.AtlasSearchIndexConfigSpec
+	ID     *string
+	Status *string
+}
+
+func (s *SearchIndex) Key() string {
```

Comment:
Do we really need `Key()`? Also, there is a `fmt.Println` here.

Reply:
No, we don't. This is a left over from testing